### DISCUSSION TITLE
error al instalar

### DISCUSSION THREAD

**Discussion by lozal2244** [Barnetik/tbai-php-lib#15] [General]
Hola, 
Muchas gracias ante todo por la aportación.

He isntalado con composer la librería y al intentar acceder a cualquier archivo de los tests me da el siguiente error:

Fatal error: Uncaught Error: Class 'PHPUnit\Framework\TestCase' not found in C:\xampp\htdocs\tbai\tests\Barnetik\Tbai\SimplifiedTicketBaiTest.php:11 Stack trace: #0 {main} thrown in C:\xampp\htdocs\tbai\tests\Barnetik\Tbai\SimplifiedTicketBaiTest.php on line 11

¿alguna idea de que falta?

Un saludo

**Comment by lozal2244** [comment]
Eliminando código e intentando avanzar me encuentro mas archivos que faltan:

Fatal error: Uncaught Error: Class 'Barnetik\Tbai\Subject\Issuer' not found in C:\xampp\htdocs\tbai\tests\Barnetik\Tbai\Mother\TicketBaiMother.php:447

**Comment by arraintxo** [comment]
No tengo muy claro cómo estás intentando acceder a los tests, lo haces a través de PHPUnit?

**Comment by lozal2244** [comment]
Buenos dias, 
Estoy accediendo a través del navegador. Estoy intentando crear un ejemplo llamando a uno de los tests para observar los resultados

**Comment by arraintxo** [comment]
Los tests no están accesibles desde el navegador, tienes que ejecutarlos usando PHPUnit desde la terminal. p.e:
```
$ vendor/bin/phpunit
```

Para testear la comunicación con las diputaciones necesitarás solicitar a cada diputación las licencias de desarrollador correspondientes y configurarlos en el phpunit.xml (copiar el phpunit.xml.dist a phpunit.xml y modifica los datos necesarios para ello)

**Comment by lozal2244** [comment]
ya tengo solicitado a cada diputación las licencias de desarrollador, pero no veo en el readme donde introducirlo, veo en el código que utliza $_ENV, pero no veo donde rellenarlo.
Nunca usé phpunit, no sería mas fácil un php con un xml que haga la llamada y desde ahi mismo ver la respuesta? por lo menos para saber que envia y cifra bien, ya luego cada uno que coja la parte del cifrado que es lo mas interesante y la adapte a su proyecto. 
Al final yo entiendoq ue PHP unit esta muy bien para testear que funciona la libreria a nivel de desarrollador, pero una persona que use el código lo primero es su Hello world. 
Por favor no lo tomen como una critica, lo digo por mejorar la librería o al menos hacerla mas atractiva, llevo ya 5 horas con este tema y se me esta atragantando un poquito ;)

**Comment by IbaiEtxebarria** [comment]
Yo tampoco consigo ejecutar los test, alguna solución

**Comment by arraintxo** [comment]
Buenas Ibai, 

Necesitaría más información para saber cual es exactamente el problema. 
¿Has pedido la licencia de desarrollo a las diputaciónes y has cambiado los datos en el phpunit.xml?

**Reply by ImPatxi** [reply]
Buenas, el error que nos da es el siguiente:

Error: Class "Test\Barnetik\Tbai\Mother\TicketBaiMother" not found

Cuando intentamos ejecutar el archivo TicketBaiTest.php con phpunit, es la primera vez que utilizamos phpunit, posiblemente no lo usemos de la manera correcta, a continuación pegó el comando que usamos:

phpunit TicketBaiTest.php

**Reply by arraintxo** [reply]
Tal y como dice @goear este error se debe a un uso incorrecto de PHPUnit o composer.
Echadle un vistazo a la documentación que os enlazan.

**Comment by goear** [comment]
https://phpunit.de/documentation.html

Creo que hay cosas y cosas, Barnetik nos brinda la oportunidad de utilizar su librería y la gente viene a preguntar como se utiliza phpunit. Ya por respeto, leed la documentación y si tenéis dudas sobre la librería de ticketbai preguntad.
